### ISSUE TITLE
import of material-design-lite/material from node_modules fails using preact build without --no-prerender

### ISSUE THREAD

**Issue by endamccormack** [developit/preact-cli#258]
I'm trying to use the [preact-mdl](https://github.com/developit/preact-mdl) with [preact-cli](https://github.com/developit/preact-cli). I'm new to preact/react so this could just be my lack of understanding but when I follow the documentation 

```
import mdl from 'material-design-lite/material';
import { Button } from 'preact-mdl';
```

this works fine in dev but when I `preact build` without `--no-prerender`(works fine with it) I get

```
system in digital-escape-velocity/ on master
› preact build
Build [==================  ] 91% (12.8s) additional asset processing
  Total precache size is about 440 kB for 17 resources.
Template execution failed: Error: ENOENT: no such file or directory, open '/Users/user/Documents/github/digital-escape-velocity/src/~/material-design-lite/material.js'
  Error: ENOENT: no such file or directory, open '/Users/user/Documents/github/digital-escap  e-velocity/src/~/material-design-lite/material.js'

  - prerender.js:69 handlePrerenderError
    [lib]/[preact-cli]/lib/lib/webpack/prerender.js:69:33

  - prerender.js:52 prerender
    [lib]/[preact-cli]/lib/lib/webpack/prerender.js:52:3

  - webpack-client-config.js:194 Object.ssr
    [lib]/[preact-cli]/lib/lib/webpack/webpack-client-config.js:194:54

  - template.html:111 1mVo.module.exports
    [lib]/[preact-cli]/lib/resources/template.html:111:37

  - index.js:264
    [lib]/[preact-cli]/[html-webpack-plugin]/index.js:264:16

  - util.js:16 tryCatcher
    [lib]/[preact-cli]/[html-webpack-plugin]/[bluebird]/js/release/util.js:16:23

  - promise.js:512 Promise._settlePromiseFromHandler
    [lib]/[preact-cli]/[html-webpack-plugin]/[bluebird]/js/release/promise.js:512:31

  - promise.js:569 Promise._settlePromise
    [lib]/[preact-cli]/[html-webpack-plugin]/[bluebird]/js/release/promise.js:569:18

  - promise.js:606 Promise._settlePromiseCtx
    [lib]/[preact-cli]/[html-webpack-plugin]/[bluebird]/js/release/promise.js:606:10

  - async.js:138 Async._drainQueue
    [lib]/[preact-cli]/[html-webpack-plugin]/[bluebird]/js/release/async.js:138:12

  - async.js:143 Async._drainQueues
    [lib]/[preact-cli]/[html-webpack-plugin]/[bluebird]/js/release/async.js:143:10

  - async.js:17 Immediate.Async.drainQueues
    [lib]/[preact-cli]/[html-webpack-plugin]/[bluebird]/js/release/async.js:17:14


entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (200 kB). This can impact web performance.
Entrypoints:
  bundle (242 kB)
      bundle.bf031.js
      style.a5e04.css

Build failed!
```

Its trying to find the file locally in the directory instead of in node_modules where it is.

**Comment by developit** [comment]
Hi @endamccormack - do you have `material-design-lite` installed?

The error actually points to node_modules (that `src/~/` is a webpack path pointing to a module lookup), but at the bottom it's failing because your build has exceeded 200kb.  Material Design Lite is a little large, and actually these days I'd strongly recommend using [preact-material-components](https://github.com/prateekbh/preact-material-components).  It's much smaller and works nicely with Preact CLI.

As a quick fix though, you can change our default `200kb` bundle size limit by adding the following to your `package.json`:

```js
{
  // package.json stuff here...
  "performance": {
    "maxAssetSize": 500000,
    "maxEntrypointSize": 500000
  }
}
```

... that being said, the limit is there to keep things speedy and it's always better to cut stuff out or push things into async chunks.

**Comment by endamccormack** [comment]
Thanks for the reply @developit. 

Yeah, that fixed the entrypoint size limit issue. I do plan on optimising when I get it working properly. Still has the issue of not finding `material-design-lite`. 

I will most likely switch out to use preact-material-components, I just don't really understand why this is failing.

Seems like its installed, its in my [package.json](https://github.com/endamccormack/digital-escape-velocity/blob/master/package.json) and its created the correct folders in node_modules by the looks of things.
```
system in digital-escape-velocity/ on master
› ls node_modules/material-design-lite
LICENSE  README.md  bower.json  dist/  gulpfile.babel.js  material.css  material.js  material.min.css  material.min.css.map  material.min.js  material.min.js.map  package.json  src/  utils/
```

**Comment by RajeshBatth** [comment]
Even I'm getting this error when using --no-prerender

$ preact build
Build [==================  ] 91% (3.7s) additional asset processing
  Total precache size is about 193 kB for 17 resources.
Template execution failed: Error: ENOENT: no such file or directory, open '/Users/Rajesh/rajesh/samples/bus-pwa-preact/src/~/invariant/invariant.js'
  Error: ENOENT: no such file or directory, open '/Users/Rajesh/rajesh/samples/bus-pwa-preact/src/~/invariant/invariant.js'

  - Error

  - prerender.js:69 handlePrerenderError
    [bus-pwa-preact]/[preact-cli]/lib/lib/webpack/prerender.js:69:33

  - prerender.js:52 prerender
    [bus-pwa-preact]/[preact-cli]/lib/lib/webpack/prerender.js:52:3

  - webpack-client-config.js:194 Object.ssr
    [bus-pwa-preact]/[preact-cli]/lib/lib/webpack/webpack-client-config.js:194:54

  - template.html:111 EM6I.module.exports
    [bus-pwa-preact]/[preact-cli]/lib/resources/template.html:111:37

  - index.js:264
    [bus-pwa-preact]/[html-webpack-plugin]/index.js:264:16

  - util.js:16 tryCatcher
    [bus-pwa-preact]/[html-webpack-plugin]/[bluebird]/js/release/util.js:16:23

  - promise.js:512 Promise._settlePromiseFromHandler
    [bus-pwa-preact]/[html-webpack-plugin]/[bluebird]/js/release/promise.js:512:31

  - promise.js:569 Promise._settlePromise
    [bus-pwa-preact]/[html-webpack-plugin]/[bluebird]/js/release/promise.js:569:18

  - promise.js:606 Promise._settlePromiseCtx
    [bus-pwa-preact]/[html-webpack-plugin]/[bluebird]/js/release/promise.js:606:10

  - async.js:138 Async._drainQueue
    [bus-pwa-preact]/[html-webpack-plugin]/[bluebird]/js/release/async.js:138:12

  - async.js:143 Async._drainQueues
    [bus-pwa-preact]/[html-webpack-plugin]/[bluebird]/js/release/async.js:143:10

  - async.js:17 Immediate.Async.drainQueues
    [bus-pwa-preact]/[html-webpack-plugin]/[bluebird]/js/release/async.js:17:14


Build failed!
error Command failed with exit code 1.


Package.json
`  "dependencies": {
    "axios": "^0.16.2",
    "history": "^4.6.3",
    "preact": "^8.2.1",
    "preact-compat": "^3.16.0",
    "react-redux": "^5.0.5",
    "react-router": "^4.1.2",
    "react-router-redux": "^4.0.8",
    "redux": "^3.7.2",
    "redux-logger": "^3.0.6",
    "redux-thunk": "^2.2.0"
  },
  "performance": {
    "maxAssetSize": 500000,
    "maxEntrypointSize": 3500000
  }`

**Comment by k1r0s** [comment]
Omg... I just finished my project and then I see this.. :'(

switching to preact-material-components then..

**Comment by diagramatics** [comment]
#317 fixes this right? This looks like a dupe of #310 and is also blocking me from using prerendering with https://github.com/prescottprue/react-redux-firebase.

**Comment by developit** [comment]
Yes. Usually there's an underlying error that the above stack trace is masking.  I'll see if we can get the current state released before we publish 2.0.
